### PR TITLE
Add WaterCluster testsystem

### PR DESCRIPTION
**Pull request by maxentile**
Creates a few waters in a harmonic restraining potential.

Implementation draws on `WaterBox` and `LennardJonesCluster`.

This was suggested by @jchodera as a testsystem that would allow a smaller number of water molecules than possible in `WaterBox`.

water_mov

### CODE REVIEW

**Comment by jchodera** [comment]
Looks like there are some travis failures:
```
======================================================================
ERROR: Failure: NameError (global name 'm' is not defined)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/loader.py", line 251, in generate
    for test in g():
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/openmmtools-0.13.5-py2.7.egg/openmmtools/tests/test_integrators_and_testsystems.py", line 88, in test_integrators_and_testsystems
    testsystem = getattr(testsystems, testsystem_name)()
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/openmmtools-0.13.5-py2.7.egg/openmmtools/testsystems.py", line 1778, in __init__
    new_pos = m.getPositions()
NameError: global name 'm' is not defined
```

**Comment by jchodera** [comment]
I've checked in a fix for the typo.

**Comment by jchodera** [comment]
For some reason, the test for `is_quantity_close()` is failing:
```
======================================================================
FAIL: Test is_quantity_close method.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/openmmtools-0.13.5-py2.7.egg/openmmtools/tests/test_utils.py", line 82, in test_is_quantity_close
    assert is_quantity_close(quantity1, quantity2) is test_result
AssertionError
```

**Comment by maxentile** [comment]
Ahh, thanks!

Looking at the `is_quantity_close` test...

**Comment by maxentile** [comment]
On all 3 Travis builds, it looks like the test that checks whether `300.0 K` is close to `300.000000004 K` is failing.

However, running `nosetests test_utils.py` locally doesn't reproduce this problem -- I wonder if the behavior of `numpy.isclose` changed recently...

**Comment by maxentile** [comment]
>I wonder if the behavior of numpy.isclose changed recently...

I think that may be the culprit. 

Conda-upgrading `numpy:        1.12.1-py36_blas_openblas_200 conda-forge [blas_openblas] --> 1.12.1-py36_blas_openblas_201 conda-forge [blas_openblas]`

Changes the behavior of
```python
import numpy as np
value1 = 300
value2 = 300.00000004
np.isclose(value1, value2, rtol=1e-10, atol=0.0) # True before, False now
np.isclose(value2, value1, rtol=1e-10, atol=0.0) # True before, False now
```

Propose we loosen the test

**Comment by maxentile** [comment]
Okay that didn't fix it-- the test is still failing on travis but passing locally...

Reverting previous commit...

File: openmmtools/tests/test_utils.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def test_is_quantity_close():
                   (1.01325*unit.bar, 1.01325000006*unit.bar, True),
                   (1.01325*unit.bar, 1.0132500006*unit.bar, False)]
     for quantity1, quantity2, test_result in test_cases:
-        assert is_quantity_close(quantity1, quantity2) is test_result
+        msg = "Test failed: ({}, {}, {})".format(quantity1, quantity2, test_result)
+        assert is_quantity_close(quantity1, quantity2) == test_result, msg
```
Comment (jchodera) [inline openmmtools/tests/test_utils.py]: NICE CATCH! That was driving me crazy.

Comment (maxentile) [reply openmmtools/tests/test_utils.py]: That was @andrrizzi ! He identified this issue in https://github.com/choderalab/openmmtools/pull/320/commits/00d2df1508000f8961ba6026d58a2c710f630532 and pointed me to the fix.

**Comment by maxentile** [comment]
Tests pass, should be good to merge.

**Comment by jchodera** [comment]
Thanks! I'll add to the docs and changelog and then merge.

**Comment by maxentile** [comment]
Oh! I forgot to do that -- no need for you to do that, I'll update the docs and changelog

**Comment by jchodera** [comment]
We should really add [pull request checklists](https://github.com/blog/2111-issue-and-pull-request-templates). I forget all the time too.

I'd add the water box [here](https://github.com/choderalab/openmmtools/blob/master/docs/testsystems.rst#water-boxes) and create a new changelog entry above [here](https://github.com/choderalab/openmmtools/blob/master/docs/releasehistory.rst) called "development release".

**Comment by jchodera** [comment]
Thanks!

**Pull request merged**